Subject: [PATCH] Avoid in6_addr redefinition

Due to both <netinet/in.h> and <linux/in6.h> being included, the
in6_addr is being redefined: once from the C library headers and once
from the kernel headers. This causes some build failures with for
example the musl C library.

In order to fix this, use just the C library header <netinet/in.h>.
Original patch taken from
http://git.alpinelinux.org/cgit/aports/tree/main/iproute2/musl-fixes.patch.

(Refreshed the patch for 4.5 release)

Upstream-Status: Pending

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Maxin B. John <maxin.john@intel.com>
---
diff -Naur iproute2-4.5.0-orig/include/libiptc/ipt_kernel_headers.h iproute2-4.5.0/include/libiptc/ipt_kernel_headers.h
--- iproute2-4.5.0-orig/include/libiptc/ipt_kernel_headers.h	2016-03-15 01:02:31.000000000 +0200
+++ iproute2-4.5.0/include/libiptc/ipt_kernel_headers.h	2016-03-23 18:33:38.579062567 +0200
@@ -6,7 +6,6 @@
 #include <limits.h>
 
 #include <netinet/ip.h>
-#include <netinet/in.h>
 #include <netinet/ip_icmp.h>
 #include <netinet/tcp.h>
 #include <netinet/udp.h>
diff -Naur iproute2-4.5.0-orig/include/linux/if_bridge.h iproute2-4.5.0/include/linux/if_bridge.h
--- iproute2-4.5.0-orig/include/linux/if_bridge.h	2016-03-15 01:02:31.000000000 +0200
+++ iproute2-4.5.0/include/linux/if_bridge.h	2016-03-23 18:33:58.569812648 +0200
@@ -15,7 +15,6 @@
 
 #include <linux/types.h>
 #include <linux/if_ether.h>
-#include <linux/in6.h>
 
 #define SYSFS_BRIDGE_ATTR	"bridge"
 #define SYSFS_BRIDGE_FDB	"brforward"
diff -Naur iproute2-4.5.0-orig/include/linux/netfilter.h iproute2-4.5.0/include/linux/netfilter.h
--- iproute2-4.5.0-orig/include/linux/netfilter.h	2016-03-15 01:02:31.000000000 +0200
+++ iproute2-4.5.0/include/linux/netfilter.h	2016-03-23 18:34:38.138317974 +0200
@@ -4,8 +4,6 @@
 #include <linux/types.h>
 
 #include <linux/sysctl.h>
-#include <linux/in.h>
-#include <linux/in6.h>
 
 /* Responses from hook functions. */
 #define NF_DROP 0
diff -Naur iproute2-4.5.0-orig/include/linux/xfrm.h iproute2-4.5.0/include/linux/xfrm.h
--- iproute2-4.5.0-orig/include/linux/xfrm.h	2016-03-15 01:02:31.000000000 +0200
+++ iproute2-4.5.0/include/linux/xfrm.h	2016-03-23 18:34:57.539075431 +0200
@@ -1,7 +1,6 @@
 #ifndef _LINUX_XFRM_H
 #define _LINUX_XFRM_H
 
-#include <linux/in6.h>
 #include <linux/types.h>
 
 /* All of the structures in this file may not change size as they are
diff -Naur iproute2-4.5.0-orig/include/utils.h iproute2-4.5.0/include/utils.h
--- iproute2-4.5.0-orig/include/utils.h	2016-03-15 01:02:31.000000000 +0200
+++ iproute2-4.5.0/include/utils.h	2016-03-23 18:35:50.066418749 +0200
@@ -1,6 +1,7 @@
 #ifndef __UTILS_H__
 #define __UTILS_H__ 1
 
+#include <sys/param.h>  /* MAXPATHLEN */
 #include <sys/types.h>
 #include <asm/types.h>
 #include <resolv.h>
